application/src/main/java/uk/gov/hmcts/dm/service/BlobStorageReadService.java

```diff
@@ -16,26 +24,117 @@
 @Transactional
 public class BlobStorageReadService {
 
+
     private final BlobContainerClient cloudBlobContainer;
+    private HttpServletRequest request;
+    private static final int DEFAULT_BUFFER_SIZE = 20480; // ..bytes = 20KB.
 
     @Autowired
-    public BlobStorageReadService(BlobContainerClient cloudBlobContainer) {
+    public BlobStorageReadService(BlobContainerClient cloudBlobContainer, HttpServletRequest request) {
         this.cloudBlobContainer = cloudBlobContainer;
+        this.request = request;
```

Comment:
it kind of feels a bit weird to have the request in the constructor and the response in the method call - can they both be given to the method?

Reply:
They can, and I actually started off that way. The main reason I changed it to remove the request was to simplify the interface as the request can be autowired (response can't) and is immutable across the requests context so no real reason to have it as an argument?

Reply:
I'd prefer it that, it just feels a bit odd. The life cycle of this object should exceed any one given request so I'm not sure how it works!